rfcs/protocols/asset-exchange/corda-htlc-alternatives.md

```diff
@@ -0,0 +1,68 @@
+<!--
+ Copyright IBM Corp. All Rights Reserved.
+
+ SPDX-License-Identifier: CC-BY-4.0
+ -->
+ 
+# Design Choices for Asset Exchange - HTLC Protocol Units in Corda Networks
+
+- RFC: 02-006
+- Authors: Ermyas Abebe, Venkatraman Ramakrishna, Sandeep Nishad, Krishnasuri Narayanam, Dhinakaran Vinayagamurthy
+- Status: Proposed
+- Since: 21-Oct-2022
+
+Here are few possible alternatives for implementing asset exchange using HTLC in a Weaver for a Corda network.
+
+## Approach Using Validating Notary
+
+Below we describe the approach for the implementation of asset exchange between Alice & Bob using HTLC that requires a validating notary.
+- Lock transaction: Consumes an input *Asset State* and produces *HTLC State* as the output. Since the claim/unlock later happens in the same ledger, a reference to the consumed *Asset State* is maintained as a Corda *StaticPointer* instead of copying the entire *Asset State* object.
+  - requires consent (signature) from both parties (Alice & Bob) for locking to complete
+<img src="../../resources/images/asset-exchange-corda-lock.png" width=80%>
+- Cliam transaction: Consumes an input *HTLC State* and produces *Asset State* as the output. Checks the validity of the hash-preimage and timout as part of a Corda contract _command_. The Corda _flow_ that implements this transaction triggers a _subFlow_ to create asset object as per _StateAndRef_ with owner set to the party claiming (Bob).
+  - The _subFlow_ will be part of the client corDapp, but not part of the interoperation corDapp of the Corda network.  
```

Comment:
General comment: capitalize "C" (and "D") in "CorDapp".

Can you make "interoperation CorDapp" link to `models/infrastructure/interoperation-modules.md`? Also, in the `interoperation-modules.md` page, there are some broken links in the last section;. Can you fix them?

Reply:
Done